xds/internal/resolver/xds_resolver.go

```diff
@@ -114,12 +113,8 @@ func (b *xdsResolverBuilder) Build(target resolver.Target, cc resolver.ClientCon
 	if err != nil {
 		return nil, err
 	}
-	endpoint := target.URL.Path
-	if endpoint == "" {
-		endpoint = target.URL.Opaque
-	}
-	endpoint = strings.TrimPrefix(endpoint, "/")
-	r.ldsResourceName = bootstrap.PopulateResourceTemplate(template, endpoint)
+	r.dataplaneAuthority = opts.Authority
```

Comment:
May I suggest:
```suggestion
	r.dataplaneAuthority = target.Endpoint()
```
... in order to honor rewritten URLs?

Reply:
Could you give me more details about what you mean by "rewritten URLs"?

Reply:
Sure: we've been using a custom resolver which translates a `custom-scheme://host` URI to `xds:///host.some-domain` ("rewrite-uri" of some sort) and then delegates to the original xDS resolver. This was working fine up to gRPC-Go 1.62.0 but, starting with gRPC-Go 1.62.1, we face:
> WARNING: [core] [Channel #1] ccResolverWrapper: reporting error to cc: no matching virtual host found for ""

The above `""` happens to come from the empty `opts.Authority`. Using `target.Endpoint()` instead makes sure the authority is gotten from the resolved target.

Both gRPC (C++) 1.62.2 and gRPC-Java 1.62.2 seem to honor "rewritten" target authorities.